tutorials/openmm-torch-nnpops.ipynb

```diff
@@ -0,0 +1,440 @@
+        "  ala2.system.removeForce(0)\n",
+        "\n",
+        "# The system should not contain any additional force and constrains\n",
+        "assert ala2.system.getNumConstraints() == 0\n",
+        "assert ala2.system.getNumForces() == 0\n",
+        "\n",
+        "# Get the list of atomic numbers\n",
+        "atomic_numbers = [atom.element.atomic_number for atom in ala2.topology.atoms()]"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "BWlQTEF1IS4S"
+      },
+      "source": [
+        "## Create a NNP\n",
+        "\n",
+        "A NNP (neural network potential) is created with [TorchANI](https://aiqm.github.io/torchani/). In this case [ANI-2x](https://chemrxiv.org/engage/api-gateway/chemrxiv/assets/orp/resource/item/60c747e9567dfec574ec48df/original/extending-the-applicability-of-the-ani-deep-learning-molecular-potential-to-sulfur-and-halogens.pdf) is used, which can be accelerated with [NNPOPs](https://github.com/openmm/NNPOps)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "ofemZo6sIbLU",
+        "outputId": "4181fcb6-db72-47e7-edb5-2a4bb0252de7"
+      },
+      "outputs": [
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "/usr/local/lib/python3.7/site-packages/torchani/__init__.py:55: UserWarning: Dependency not satisfied, torchani.ase will not be available\n",
+            "  warnings.warn(\"Dependency not satisfied, torchani.ase will not be available\")\n"
+          ]
+        },
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "/usr/local/lib/python3.7/site-packages/torchani/resources/\n",
+            "Downloading ANI model parameters ...\n"
+          ]
+        },
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "/usr/local/lib/python3.7/site-packages/torch/functional.py:1069: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at  /home/conda/feedstock_root/build_artifacts/pytorch-recipe_1640851336451/work/aten/src/ATen/native/TensorShape.cpp:2156.)\n",
+            "  return _VF.cartesian_prod(tensors)  # type: ignore[attr-defined]\n"
+          ]
+        }
+      ],
+      "source": [
+        "import torch as pt\n",
+        "from torchani.models import ANI2x\n",
+        "from NNPOps import OptimizedTorchANI\n",
+        "\n",
+        "class NNP(pt.nn.Module):\n",
+        "\n",
+        "  def __init__(self, atomic_numbers):\n",
+        "\n",
+        "    super().__init__()\n",
+        "\n",
+        "    # Store the atomic numbers\n",
+        "    self.atomic_numbers = pt.tensor(atomic_numbers).unsqueeze(0)\n",
```

Comment:
How does this work for atoms that should _not_ have their potential computed by the NNP? Would we use atomic numbers of `0` for those atoms? For example, if we wanted to only compute ligand energies for a solvated protein:ligand system?

Reply:
I think that's beyond the scope of this first tutorial.  Mixing ML with MM could be a more advanced tutorial.  Or we could just refer people to OpenMM-ML, which has code to automate it.

Reply:
It seems weird not to even mention how it would work, since this is a major use case. If folks wanted to just simulate only ML, why would they use OpenMM? 

I agree a separate tutorial would be useful!

What about periodic ML systems? Do we support those? Or just non-periodic systems?

Maybe we could have three tutorials:
* Vacuum ML
* Periodic ML
* Hybrid ML/MM periodic

Reply:
Periodic systems just take a minor change to the model.  See https://github.com/openmm/openmm-torch/blob/master/README.md.